crates/store/src/state.rs

```diff
@@ -800,6 +801,20 @@ impl State {
 
         let account_commitment = inner.account_tree.get(account_id);
 
+        let new_account_id_prefix_is_unique = if account_commitment.is_empty() {
+            Some(!inner.account_tree.contains_account_id_prefix(account_id.prefix()))
+        } else {
+            None
+        };
+
+        // Non-unique account Id prefixes for new accounts are not allowed.
+        if let Some(false) = new_account_id_prefix_is_unique {
+            return Ok(TransactionInputs {
+                new_account_id_prefix_is_unique,
+                ..Default::default()
+            });
+        }
+
```

Comment:
I think its easier to grok if the early return is embedded, but up to you.

```suggestion
        if account_commitment.is_empty() && 
            !inner.account_tree.contains_account_id_prefix(account_id.prefix()))
         {
            return Ok(TransactionInputs {
                new_account_id_prefix_is_unique,
                ..Default::default()
            });
        }
```

And then set the `Ok` variant at the end to `None`/`Default`.